### ISSUE TITLE
BBBGetLatestAMI not working

### ISSUE THREAD

**Issue by klasuk** [aws-samples/aws-scalable-big-blue-button-example#118]
**Description**

Creating stack stopped with error msg: "Failed to create/update the stack".
Failed task is actually function BBBGetLatestAMI (from template bbb-on-aws-amifinder.template.yaml). Error message is like:
Embedded stack arn:aws:cloudformation:eu-north-1:_[myawsuser]_:stack/mmgbbb-BBBAMIFinderStack-3URKTNCHBYHH/5e3d6100-3a57-11ed-b0b6-0a84cc47e266 was not successfully created: The following resource(s) failed to create: [BBBGetLatestAMI].

I just cannot understand what is wrong with this python function. Supported Ubuntu versions and platform should be correct.

**Comment by couchgott** [comment]
Hi @klasuk 

can you take a loog at the nested stack details/messages. there should be a hint what exactly went wrong. 

Ty!

**Comment by klasuk** [comment]
I'm using latest versio (tag=4.0.0).

setup -script stopped with error:

##################################################
Registry Preperation finished
Building the BBB Environment
##################################################

Waiting for changeset to be created..
Waiting for stack create/update to complete

Failed to create/update the stack. Run the following command
to fetch the list of events leading up to the failure
aws cloudformation describe-stack-events --stack-name mmgbbb
##################################################
Deployment finished

Stack error log attached:

[stack_error.log](https://github.com/aws-samples/aws-scalable-big-blue-button-example/files/9684616/stack_error.log)

**Comment by couchgott** [comment]
From what i See in the Log you send your Network Stack and particular the Nat Gateway did fail to be deployed. 

        {
            "StackId": "arn:aws:cloudformation:eu-north-1:[myawsaccountid]:stack/mmgbbb/b066c940-40c3-11ed-9e51-0663462825c2",
            "EventId": "BBBNetworkStack-CREATE_FAILED-2022-09-30T13:29:12.521Z",
            "StackName": "mmgbbb",
            "LogicalResourceId": "BBBNetworkStack",
            "PhysicalResourceId": "arn:aws:cloudformation:eu-north-1:[myawsaccountid]:stack/mmgbbb-BBBNetworkStack-1I30B0NYY4ZPD/ba5f46c0-40c3-11ed-af50-0649b99e3ac4",
            "ResourceType": "AWS::CloudFormation::Stack",
            "Timestamp": "2022-09-30T13:29:12.521000+00:00",
            "ResourceStatus": "CREATE_FAILED",
            "ResourceStatusReason": "Embedded stack arn:aws:cloudformation:eu-north-1:[myawsaccountid]:stack/mmgbbb-BBBNetworkStack-1I30B0NYY4ZPD/ba5f46c0-40c3-11ed-af50-0649b99e3ac4 was not successfully created: The following resource(s) failed to create: [BBBNatGateway1EIP]. ",
            "R

Can you please check this stacks Events in detail (Details for each resource). Maybe using the UI works better here. 
It seems something on your VPC setup is not correct or causing issues.

**Comment by klasuk** [comment]
I managed to resolve network issue (by deleting previous vpc networks manually) and deployed again. I've checked bbb-params (json) file several times. I noticed that BBBDBEngineVersion should be currently 10.18 (at least) or newer:
https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraPostgreSQL.Updates.20180305.html

Deployment seems to get redis cluster created but creation of Aurora DB cluster failed:
Embedded stack arn:aws:cloudformation:eu-north-1:[myawsid]:stack/mmgbbb-BBBDatabaseStack-J28DUYQMPQ9L/1c731c30-431a-11ed-b86a-06fc7f6a6170 was not successfully created: The following resource(s) failed to create: [BBBRDSCluster].

**Comment by couchgott** [comment]
okay. same as before, check for the error at the nested stack there should be a hint at the events what exactly is causing the error. 
yes the default parameter might be outdated. this will be corrected in the next release/update of the code-stack. 
til then can be easily corrected changing that parameter at the file.

**Comment by klasuk** [comment]
I noticed that you updated repo with latest  fixes (and with BBB 2.5). These are almost same fixes that I figured out with my own testing... :) Thank you for returning db cluster back to serverless (It didn't work as serverless inside eu-north-1... but now that seems to work again).

I have still problem with turn stack:

You must use a valid fully-formed launch template. The requested configuration is currently not supported. Please check the documentation for supported configurations. (Service: AmazonAutoScaling; Status Code: 400; Error Code: ValidationError; Request ID: f99afdf1-948a-4c4c-99a8-fc9db2eff695; Proxy: null)

I tried to fix this template scheme throw offical aws documentation without any luck. Could this be amazon zone ralated issue?

**Comment by couchgott** [comment]
hmm not sure, i can try to deplyo to eu-north-1 as well and look how it is doing
currently i am working on a last action item with the code to the greenlight deployments adjustable as a parameter (for folks using the deployment with e.g. moodle) 
give me some time and i'll come back here

**Comment by couchgott** [comment]
yep i tried in eu-north-1 and got that error as well. i did not get it in eu-central-1 deployments so far. intresting

**Comment by couchgott** [comment]
soo. i tried 2 additional regions without an issue. let me reach out internally and find out what is causing this. 
i will do a release of the current state in any case in a few minutes and update you here if i have an idea/mitigation to get the stack deploying in the Stockholm region. Sorry for the inconvenience

**Comment by couchgott** [comment]
okay. so. @klasuk you might want to check the instance types you are using. as default i set t3a instance families within the parameter file. t3a instances are not available in eu-north-1 so you should change the parameter to t3. and then it should work! 

let me know ;) 

Best.
